WordPress/Classes/ViewRelated/Tools/PromptViewController.swift

```diff
@@ -0,0 +1,129 @@
+import Foundation
+import UIKit
+
+
+/// ViewController container, that presents a Done / Cancel button, and forwards their events to
+/// the childrenViewController (which *must* implement the Confirmable protocol).
+///
+class PromptViewController : UIViewController
+{
+    // MARK: - Static Helpers
+    
+    /// Wraps a given UIViewController, that conforms to the Confirmable Protocol, into:
+    /// -   A PromptViewController instance, which deals with the NavigationItem buttons
+    /// -   (And verything) inside a UINavigationController instance.
+    ///
+    static func navigationControllerWithPrompt<T: UIViewController where T: Confirmable>(viewController: T) -> UINavigationController {
+        let viewController = PromptViewController(viewController: viewController)
+        return UINavigationController(rootViewController: viewController)
+    }
```

Comment:
I kind of like in the previous version that the actual prompt view controller was private to this file. With this version you could still mistakenly initialize a `PromptViewController` directly.

I'd go back to having a private class, but having a top level public constructor function instead of the navigation controller subclass. Then there's only one possible way to instantiate this and no room for error.

```swift
public func PromptViewController<T: UIViewController where T: Confirmable>(viewController: T) -> UINavigationController {
    let viewController = PromptContainerViewController(viewController: viewController)
    return UINavigationController(rootViewController: viewController)
}
```